pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go

```diff
@@ -49,17 +50,20 @@ type CollectorBundle struct {
 // If that's not the case then it'll select all available collectors that are
 // marked as stable.
 func NewCollectorBundle(chk *OrchestratorCheck) *CollectorBundle {
+	mb := NewManifestBuffer(chk)
 	bundle := &CollectorBundle{
 		discoverCollectors: chk.orchestratorConfig.CollectorDiscoveryEnabled,
 		check:              chk,
 		inventory:          inventory.NewCollectorInventory(),
 		runCfg: &collectors.CollectorRunConfig{
-			APIClient:   chk.apiClient,
-			ClusterID:   chk.clusterID,
-			Config:      chk.orchestratorConfig,
-			MsgGroupRef: chk.groupID,
+			APIClient:         chk.apiClient,
+			ClusterID:         chk.clusterID,
+			Config:            chk.orchestratorConfig,
+			MsgGroupRef:       chk.groupID,
+			ManifestBufferCfg: mb.Cfg,
```

Comment:
I couldn't find a usage of this in the CollectorBundle + its already in the manifestBuffer below? Could you show me where we need the split and duplicate?

Reply:
Yeah this is a little bit redundant, I did this to simplify the code structure .

Here we pass CollectorBundle.runCfg to collector, 
https://github.com/DataDog/datadog-agent/blob/ab3caf41d816ed85c09334a7e066165b0df69c6e/pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go#L236

In each collector, we generate processor context
https://github.com/DataDog/datadog-agent/blob/ab3caf41d816ed85c09334a7e066165b0df69c6e/pkg/collector/corechecks/cluster/orchestrator/collectors/collector.go#L125-L135

And eventually in the processor, we need to use `BufferedManifestEnabled` and `ManifestChan`

Reply:
Ah I see, but wdyt about removing that from the processor altogether and moving all of that in the collectorBundle.Run(). Once you receive the processorResult you could loop through it and send that to the channel. With this the processor and each resource itself is independent/unaware of the manifest buffering and only the collectorBundle is aware of that.

Reply:
Ah I see, I am a bit torn between:
- keeping your current implementation
- or removing that from the processor altogether and moving all of that in the collectorBundle.Run(). Once you receive the processorResult you could loop through it and send that to the channel. With this the processor and each resource itself is independent/unaware of the manifest buffering and only the collectorBundle is aware of that. 

The first does not do redundant work (the chunking) + saving the message in the slice
The former is cleaner due to separation

wdyt?

Reply:
We can probably passe one new parameters to processor `ChunkManifest : bool` instead of `BufferedManifestEnabled / ManifestChan`

We can let the processor return chunked metadata payloads and unchunked manifests and in collectorbundle we loop through them and send to the channel

Reply:
I eventually choose the second solution removing everything to the ` collectorBundle.Run()` which makes the code much easier to read and the redundant work is not that costly